app.py

```diff
@@ -3,36 +3,46 @@
 from datetime import date
 import os
 import uuid
-from flask import render_template, request, session, redirect, Flask
-from flask_login import login_user, current_user, LoginManager
+from flask import render_template, request, redirect, Flask, url_for
+from flask_login import login_user, logout_user, login_required, current_user
+from flask_login import LoginManager
 from src.db_models import User, UserActivity, UserHabit, db
 from src.db_manager import DBManager
 from src.utils import AlchemyEncoder
 
 # Flask App init
 app = Flask(__name__)
-basedir = os.path.abspath(os.path.dirname(__file__))
-app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = True
-app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///' + \
-    os.path.join(basedir, 'db.sqlite')
 app.config['SECRET_KEY'] = uuid.uuid4().hex
-db.init_app(app)
 app.app_context().push()
 
+# login
 login_manager = LoginManager()
 login_manager.init_app(app)
+login_manager.login_view = "/"
+login_manager.login_message = u"you must login to access this page"
+login_manager.login_message_category = "message"
+
 
+# db
+db.init_app(app)
 db.create_all()
+# TODO using a local db.session might be a better idea
 db_manager = DBManager(db.session)
```

Comment:
I think this global db.session is ok??? FlaskLogin does a bunch of scoped local session under the hood